src/Symfony/Component/Form/Tests/ResolvedFormTypeTest.php

```diff
@@ -179,6 +179,33 @@ public function testCreateBuilderWithDataClassOption()
         $this->assertSame('\stdClass', $builder->getDataClass());
     }
 
+    /**
+     * @expectedException \Symfony\Component\OptionsResolver\Exception\MissingOptionsException
+     * @expectedExceptionMessage An error has occurred resolving the options of the form "stdClass": The required option "foo" is missing.
+     */
+    public function testFailsCreateBuilderOnInvalidFormOptionsResolution()
+    {
+        $optionsResolver = (new OptionsResolver())
+            ->setRequired('foo')
+        ;
+        $this->resolvedType = $this->getMockBuilder('Symfony\Component\Form\ResolvedFormType')
```

Comment:
let's use ::class
or better: not use any mocks if possible :)

Reply:
done via `::class`